### ISSUE TITLE
Fix issues from Pagination Axe Scan

### ISSUE THREAD

**Issue by rsmithadhoc** [department-of-veterans-affairs/vets-design-system-documentation#2111]
## Description

Correct an Axe issue with `va-pagination` found during accessibility review. 

## Details

- **Must** _List element has direct children that are not allowed: `[role=presentation]`_
    - Related Node
    ```html
    <li class="usa-pagination__item usa-pagination__overflow" aria-label="ellipsis indicating non-visible pages" role="presentation"><span>...</span></li>
    ```
    - <img width="400" alt="image" src="https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/100248909/6f49cc7f-9f1a-4b09-abd2-757e2649fc29">
    - **Recommendation**: Remove `role=presentation`. The presentation attribute hides that element's semantics from assistive technology. For this `<li>`, it would no longer be considered a list item. I believe it should be a list item, so those using assistive technology are able to see that there is a gap in page numbers. See [ARIA: presentation role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role) and [axe-core 4.6](https://dequeuniversity.com/rules/axe/4.6/list) for more information.

## Tasks
- [ ] Fix the accessibility issue

## Acceptance Criteria
- [ ] Accessibility issue addressed
